File: files/400-499/409/CH-409.R
library(tidyverse)
library(readxl)

path <- "400-499/409/CH-409 Filter.xlsx"
input <- read_excel(path, range = "B3:B10")
test <- read_excel(path, range = "F3:F5")

result <- input %>%
  mutate(num = str_extract_all(ID, "\\d+")) %>%
  unnest(num) %>%
  filter(
    as.numeric(str_sub(num, 1, 1)) %% 2 == 0,
    as.numeric(str_sub(num, -1, -1)) %% 2 == 1
  )

all.equal(result$ID, test$ID)
# [1] TRUE

File: files/400-499/409/CH-409.py
import pandas as pd

path = "400-499/409/CH-409 Filter.xlsx"
input = pd.read_excel(path, usecols="B", nrows=8, skiprows=2)
test = pd.read_excel(path, usecols="F", nrows=2, skiprows=2)
result = (
	input.assign(num=input.ID.str.findall(r"\d+"))
	.explode("num")
	.loc[
		lambda d: (d.num.str[0].astype(int) % 2 == 0)
		& (d.num.str[-1].astype(int) % 2 == 1)
	]
)

print(result["ID"].tolist() == test["ID.1"].tolist())
# Output: True